doc/book/reference/migration/to-v1.1.md

```diff
@@ -0,0 +1,560 @@
+use Zend\Expressive\Middleware\ErrorResponseGenerator;
+
+return [
+    'dependencies' => [
+        'factories' => [
+            ErrorResponseGenerator::class => WhoopsErrorResponseGeneratorFactory::class,
+        ],
+    ],
+];
+```
+
+Other things you may want to do:
+
+- The `ErrorHandler` entry could potentially be moved inwards a few layers. As
+  an example, the `ServerUrlMiddleware` has no possibility of raising an
+  exception or error, and could be moved outwards; you could do similarly for
+  any middleware that only injects additional response headers.
+
+- Remove any Stratigility-style error middleware (middleware expecting an error
+  as the first argument). If any specialized error handling should occur, add
+  additional middleware into the pipeline that can catch exceptions, and have
+  that middleware re-throw for exceptions it cannot handle.
+
+- Consider providing your own `Zend\Stratigility\NoopFinalHandler`
+  implementation; this will now only be invoked if the queue is exhausted, and
+  could return a generic 404 page, raise an exception, etc.
+
+## http-interop
+
+Stratigility 1.3 provides the ability to work with [http-interop middleware
+0.2.0](https://github.com/http-interop/http-middleware/tree/ff545c87e97bf4d88f0cb7eb3e89f99aaa53d7a9).
+
+This specification, which is being developed as the basis of
+[PSR-15](https://github.com/php-fig/fig-standards/tree/master/proposed/http-middleware),
+defines what is known as _lambda_ or _single-pass_ middleware, vs the
+_double-pass_ middleware traditionally used by Stratigility and Expressive.
+
+Double-pass refers to the fact that two arguments are passed to the delegation
+function `$next`: the request and response. Lambda or single-pass middleware
+only pass a single argument, the request.
+
+Stratigility 1.3 provides support for dispatching either style of middleware.
+
+Specifically, your middleware can now implement:
+
+- `Interop\Http\Middleware\ServerMiddlewareInterface`, which defines a single
+  method, `process(ServerRequestInterface $request,
+  Interop\Http\Middleware\DelegateInterface $delegate)`.
+- Callable middleware that follows the above signature (the typehint for the
+  request argument is optional).
+  
+Both styles of middleware may be piped directly to the middleware pipeline or as
+routed middleware within Expressive. In each case, you can invoke the
+next middleware layer using `$delegate->process($request)`.
+
+Starting in Stratigility 2.0 and Expressive 2.0, `Application` will continue to
+accept the legacy double-pass signature, but will require that you either:
+
+- Provide a `$responsePrototype` (a `ResponseInterface` instance) to the
+  `Application` instance prior to piping or routing such middleware.
+- Decorate the middleware in a `Zend\Stratigility\Middleware\CallableMiddlewareWrapper`
+  instance (which also requires a `$responsePrototype`).
+
+We recommend that you begin writing middleware to follow the http-interop
```

Comment:
Maybe add an example here to clarify it a bit. I think it's something like this? (Disclaimer: from memory and untested)

```php
<?php

namespace App\Middleware;

use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface;
use Interop\Http\Middleware\ServerMiddlewareInterface;
use Interop\Http\Middleware\DelegateInterface;

class MyMiddleware implements ServerMiddlewareInterface
{
    public function __construct()
    {
    }

    /**
     * @param Request           $request
     * @param DelegateInterface $delegate
     *
     * @return Response
     */
    public function process(ServerRequestInterface $request, DelegateInterface $delegate): ResponseInterface
    {
        // Do something (with the request) first

        // Call the next middleware and wait for the response
        $response = $delegate->process($request);

        // Do something (with the response) before returning the response

        // Return the response
        return $response;
    }
}
```